src/test/java/com/google/firebase/auth/UserImportHashTest.java

```diff
@@ -145,6 +146,91 @@ public void testBasicHash() {
     }
   }
 
+  private void assertBuilderThrowsIllegalArgumentException(Md5.Builder builder) {
```

Comment:
An alternative to these helpers is to make all of the builders extend from a common interface (which they do: RepeatableHash.Builder) and make it public (which it isn't; and neither is RepeatableHash) even if it's just `@VisibleForTesting`. I didn't do that, in the (naively?) optimistic hope that we'll update to java8 "soon" which will make all that unnecessary.

Reply:
May be move this to `InvalidHashTest`. There `RepeatableHash` is already visible.

Reply:
Oh, yeah; that's definitely better. (Despite the name, I also used this file to test for valid hashes too.)